tests/StorageTrait.php

```diff
@@ -0,0 +1,263 @@
+            $isSliced,
+            $isDirectory,
+            $primaryKeys
+        );
+    }
+
+    /**
+     * @param string[] $columns
+     * @param string[]|null $primaryKeys
+     * @return S3\SourceFile|ABS\SourceFile|S3\SourceDirectory|ABS\SourceDirectory
+     */
+    public function getSourceInstanceFromCsv(
+        string $filePath,
+        CsvOptions $options,
+        array $columns = [],
+        bool $isSliced = false,
+        bool $isDirectory = false,
+        ?array $primaryKeys = null
+    ) {
+        switch (getenv('STORAGE_TYPE')) {
+            case StorageType::STORAGE_S3:
+                $getSourceInstanceFromCsv = 'createS3SourceInstanceFromCsv';
+                $manifestPrefix = 'S3.';
+                break;
+            case StorageType::STORAGE_ABS:
+                $getSourceInstanceFromCsv = 'createABSSourceInstanceFromCsv';
+                $manifestPrefix = '';
+                break;
+            default:
+                throw new \Exception(sprintf('Unknown STORAGE_TYPE "%s".', getenv('STORAGE_TYPE')));
+        }
+
+        $filePath = str_replace('%MANIFEST_PREFIX%', $manifestPrefix, $filePath);
+        return $this->$getSourceInstanceFromCsv(
+            $filePath,
+            $options,
+            $columns,
+            $isSliced,
+            $isDirectory,
+            $primaryKeys
+        );
+    }
+
+    public function clearDestination(string $dirToClear): void
+    {
+        switch (getenv('STORAGE_TYPE')) {
+            case StorageType::STORAGE_S3:
+                /** @var S3Client $client */
+                $client = $this->createClient();
+                $result = $client->listObjects([
+                    'Bucket' => (string) getenv('AWS_S3_BUCKET'),
+                    'Prefix' => $dirToClear,
+                ]);
+                $objects = $result->get('Contents');
+                if ($objects) {
+                    $client->deleteObjects([
+                        'Bucket' => (string) getenv('AWS_S3_BUCKET'),
+                        'Delete' => [
+                            'Objects' => array_map(static function ($object) {
+                                return [
+                                    'Key' => $object['Key'],
+                                ];
+                            }, $objects),
+                        ],
+                    ]);
+                }
+                return;
+            case StorageType::STORAGE_ABS:
+                /** @var BlobRestProxy $client */
+                $client = $this->createClient();
+                // delete blobs from EXPORT_BLOB_DIR
+                $listOptions = new ListBlobsOptions();
+                $listOptions->setPrefix($dirToClear);
+                $blobs = $client->listBlobs((string) getenv('ABS_CONTAINER_NAME'), $listOptions);
+                foreach ($blobs->getBlobs() as $blob) {
+                    $client->deleteBlob((string) getenv('ABS_CONTAINER_NAME'), $blob->getName());
```

Comment:
ten env bych si vytahl do promenne pred foreachem at se to nemusi volat Xkrat

Reply:
V testech je to asi fuk, ale proč ne.